src/core.c/CompUnit/Repository/FileSystem.pm6

```diff
@@ -320,11 +323,12 @@ class CompUnit::Repository::FileSystem
           .subst(:g, /\:\:+/, '::')
           .subst(/^.*?'::'/, '')
           .subst(/\..*/, '')
-        }{ .IO.extension } = $_
-          for ls($!prefix.absolute).grep(*.ends-with(any(@extensions)));
+        }{ '.' ~ .IO.extension } = $_
+          for ls($!prefix.absolute).grep(*.ends-with(any(@!extensions)));
```

Comment:
I just tested `raku -I . -e 'use Test'` in a directory with ~130k directories/files. Stock rakudo took ~8.6s, but changing the `*.ends-with(any(@!extensions)` to `{ .ends-with('.rakumod') || .ends-with('.pm6') || .ends-with('.pm') }` dropped it to ~7.6s. FWIW, the function at the top of a `--profile-compile` report is `abs2rel`.

Reply:
I'll update CURFS to remove the use of junctions after I merge this to avoid the conflict on `extensions`

Reply:
`abs2rel` could be called less. Instead of creating the relative paths inside of `ls(...)`, they could be created after the call to ls (i.e. `ls(...).map(*.relative(...))`. The reason to do this is it avoids creating relative paths for files that don't match any of the extensions we consider raku modules.